File: TransCoder_tokenized_test_set_functions/python/FIND_MEDIAN_ROW_WISE_SORTED_MATRIX.py
def binaryMedian ( m, r, d ) :
    mi = m [ 0 ] [ 0 ]
    mx = 0
    for i in range ( r ) :
        if m [ i ] [ 0 ] < mi :
            mi = m [ i ] [ 0 ]
        if m [ i ] [ d - 1 ] > mx :
            mx = m [ i ] [ d - 1 ]
    desired = ( r * d + 1 ) //2
    while(mi<mx):
        mid=mi+(mx-mi)//2
        place=[0];
        foriinrange(r):
            j=upper_bound(m[i],mid)
            place[0]=place[0]+j
        ifplace[0]<desired:
            mi=mid+1
        else:
            mx=mid
    print("Median is",mi)
    return

File: results/java_codex/FIND_MEDIAN_ROW_WISE_SORTED_MATRIX.java
int binaryMedian(int m[][], int r, int c) {
        int max = INT_MIN;
        int min = INT_MAX;
        for (int i = 0; i < r; i++) {
            if (m[i][0] < min) min = m[i][0];
            if (m[i][c - 1] > max) max = m[i][c - 1];
        }
        int desired = (r * c + 1) / 2;
        while (min < max) {
            int mid = min + (max - min) / 2;
            int place = 0;
            int get = 0;
            for (int i = 0; i < r; ++i) {
                get = binarySearch(m[i], mid);
                if (get < 0) get = abs(get) - 1;
                else {
                    while (get < m[i].length && m[i][get] == mid) get += 1;
                }
                place = place + get;
            }
            if (place < desired) min = mid + 1;
            else max = mid;
        }
        return min;
    }